src/main/java/core/basesyntax/strategy/impl/BalanceOperation.java

```
@@ -0,0 +1,15 @@
+package core.basesyntax.strategy.impl;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.strategy.OperationHandler;
+
+public class BalanceOperation implements OperationHandler {
+    @Override
+    public void handle(FruitTransaction transaction) {
+        if (transaction == null) {
+            throw new RuntimeException("Transaction cannot be null.");
+        }
+        Storage.setFruitQuantity(transaction.getFruit(), transaction.getQuantity());
```

handle(...) checks only that `transaction` is non-null but does not validate `transaction.getFruit()` (non-null/non-empty) nor `transaction.getQuantity()` (non-negative). The checklist requires anticipating invalid inputs and failing fast with informative RuntimeExceptions. Add checks such as:
`if (transaction.getFruit() == null || transaction.getFruit().trim().isEmpty()) { throw new RuntimeException("Fruit name cannot be null or empty."); }`
`if (transaction.getQuantity() < 0) { throw new RuntimeException("Quantity cannot be negative: " + transaction.getQuantity()); }`
Perform these before calling Storage.setFruitQuantity(...). See DataConverterImpl for similar validation at parse time (do not rely solely on it).